src/test/clojure/clara/test_rules.clj

```diff
@@ -2675,3 +2701,25 @@
 
       (is (= {:?w 10, :?t nil}
              (-> e (ex-data) :bindings))))))
+
+;;; Test that we can properly assemble a session, insert facts, fire rules,
+;;; and run a query with keyword-named productions.
+(deftest test-simple-insert-data-with-keyword-names
+  (let [session (-> (mk-session (rules-data/weather-rules-with-keyword-names))
+                    (insert (->Temperature 15 "MCI"))
+                    (insert (->WindSpeed 45 "MCI"))
+                    (fire-rules))]
+    (is (= [{:?fact (->ColdAndWindy 15 45)}]
+           (query session ::rules-data/find-cold-and-windy-data)))))
+
+;;; Verify that an exception is thrown when a duplicate name is encountered.
+;;; Note we create the session with com/mk-session*, as com/mk-session allows
+;;; duplicate names and will take what it considers to be the most recent
+;;; definition of a production.
+(deftest test-duplicate-name
+  (try
```

Comment:
Let's use assert-ex-data instead.  That has the benefit of, among other things being less brittle if another layer of try-catch is added.  I think I just didn't notice this before, sorry for the late changes.